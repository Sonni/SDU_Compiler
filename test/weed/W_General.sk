var hej: int;
var other: int;

func f(): int
    var a: int;
    return 0;
    a = 40;
    a = 10;
    return 10;
end f

if (true || false) && (true && false) then
{
	hej=1;
} 

if !(other > 10 && hej == other) then
{
	hej=2;
	hej=3+4*2;
} 

if (2 == 2 || hej == 7) then 
{
	hej=3;
} 

while false do
{
	hej=4;
}

if (2 == 3) then
{
	hej=5;
}
else
{
	hej=6;
}

if (2 == 2) then
{
	hej=5;
}
else
{
	hej=6;
}

while true do
{
    break;
	hej=4;
}

write 0;