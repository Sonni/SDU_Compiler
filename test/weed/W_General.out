Test by visual inspect.. compile with: ./compiler -p n
to pretty print the AST and exit after

func f(): int
    var a: int;
    return 0;
end f

if other <= 10 || hej != other then
{
	hej=2;
	hej=11;
}

hej=3;
hej=6;
hej=5;

while true do
{
    break;
}

write 0;